internal/cmd/reading.go

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+var readingDevice string
+var readingLimit, readingOffset int
+
+func init() {
+	readingCmd := initReadingCommand()
+	initListReadingCommand(readingCmd)
+	initCountReadingCommand(readingCmd)
+}
+
+func initReadingCommand() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:          "reading",
+		Short:        "Count and list readings",
+		Long:         ``,
+		SilenceUsage: true,
+	}
+	rootCmd.AddCommand(cmd)
+	addFormatFlags(cmd)
+	return cmd
+}
+
+func initListReadingCommand(cmd *cobra.Command) {
+	var listCmd = &cobra.Command{
+		Use:          "list",
+		Short:        "List all readings",
+		Long:         `List all readings, optionally specifying a limit and offset`,
```

Comment:
I think it's import to note that this command returns a list of reading from newest to oldest. I tried the limit and offset args and although they work, it seems counter-intuitive to page through the list in this order, especially if the system is constantly generating new readings. Perhaps this endpoint needs an additional parameter which would allow the order to specified? @lenny-intel ^^